pkg/recipes/driver/bicep.go

```diff
@@ -162,6 +163,12 @@ func (d *bicepDriver) Execute(ctx context.Context, opts ExecuteOptions) (*recipe
 
 	resp, err := poller.PollUntilDone(ctx, &runtime.PollUntilDoneOptions{Frequency: pollFrequency})
 	if err != nil {
+		errorDetails := recipes.GetErrorDetails(err)
+		for _, error := range errorDetails.Details {
+			if error.Code == "AuthorizationFailed" && providerConfig.Az == nil {
+				return nil, v1.NewClientErrInvalidRequest("you are deploying Azure resource but the environment does not have a subscription reference.")
```

Comment:
Should we add a test for this? I guess, the message can be clearer. cc/ @willtsai

Reply:
no, we cannot add a unit test for this, as we are making a ucp call here and we dont mock it.

Reply:
how about:
`Azure deployment failed due to authorization error, please ensure you have registered an Azure provider with your Radius environment: https://docs.radapp.io/guides/operations/providers/azure-provider/`